src/Features/CSharp/Portable/ImplementInterface/AbstractChangeImplementionCodeRefactoringProvider.cs

```diff
@@ -69,7 +69,10 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
             // member.  Interface member names are the expected names that people expect to see
             // (like "GetEnumerator"), instead of the auto-generated names that the compiler makes
             // like: "System.IEnumerable.GetEnumerator"
-            directlyImplementedMembers.AddRange(member, member.ExplicitOrImplicitInterfaceImplementations());
+            var interfaceImplementations = member.ExplicitOrImplicitInterfaceImplementations();
+            if (interfaceImplementations.Length == 0)
```

Comment:
Would it be clearer to add logic to `CheckMemberCanBeConverted` that checks if the interface appears in the `member.ContainingType.AllInterfaces` list?

This feels a little like this is working by side effect of `ExplicitOrImplicitInterfaceImplementations` iterating through the interfaces.

Reply:
Yes it might make things a bit clearer.

@CyrusNajmabadi what do you think?

Reply:
Ok so I went ahead with that idea anyway. I had to duplicate code though, since `CheckMemberCanBeConverted` lies in the concrete classes.

Reply:
works for me :)

Reply:
Done.

I feel the code is now self-documenting enough. Do you? 🤔